qualtran/surface_code/ui.py

```diff
@@ -522,24 +522,30 @@ def update(
     qec_name: str,
     magic_name: str,
     magic_count: int,
-    rotaion_model_name: str,
+    rotation_model_name: str,
 ):
     """Updates the visualization."""
     if any(x is None for x in [physical_error_rate, error_budget, *algorithm_data, magic_count]):
         raise PreventUpdate
 
     # TODO: We implicitly rely on the order of the input components
     qubits, measurements, ts, toffolis, rotations, n_rotation_layers = algorithm_data
+
+    rotation_eps = 1e-3 / rotations  # TODO this should not be a magic number
```

Comment:
should we support "unknown precision" rotations. I guess in the rotation bloqs we have a default eps; but it's not hard to imagine a situation where a researcher has a bunch of rotations and they haven't computed an epsilon for each of them

Reply:
I'm using `error_budget / rotations` here